### PR TITLE
[buildessential] update gcc version to 7.3.0

### PR DESCRIPTION
**Pull request by lovedheart**

### CODE REVIEW

File: packages/buildessential.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class Buildessential < Package
 
   is_fake
 
-  depends_on 'gcc'
+  depends_on 'gcc7'
```
Comment (uberhacker) [inline packages/buildessential.rb]: You sure we're ready for this?  Bump version?

Comment (lovedheart) [reply packages/buildessential.rb]: The option gcc4-compatible is added, no need to compile the packages from source code again.

Comment (uberhacker) [reply packages/buildessential.rb]: Let's test this a bit more.  Please compile some more packages with gcc7 and push a few PRs.

Comment (uberhacker) [reply packages/buildessential.rb]: Tests results good.  Nice work @lovedheart!

**Pull request merged**